### DISCUSSION TITLE
How to get RefreshToken

### DISCUSSION THREAD

**Discussion by bennywoop** [abuzuhri/Amazon-SP-API-CSharp#23] [Q&A] [answered]
How to get RefreshToken here:

AmazonConnection amazonConnection = new AmazonConnection(new AmazonCredential()
    {
         AccessKey = "AKIAXXXXXXXXXXXXXXX",
         SecretKey = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
         RoleArn = "arn:aws:iam::XXXXXXXXXXXXX:role/XXXXXXXXXXXX",
         ClientId = "amzn1.application-XXX-client.XXXXXXXXXXXXXXXXXXXXXXXXXXXX",
         ClientSecret = "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX",
          **RefreshToken**= "XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX"
     });

Do I need to request here? Or do you have a function to call in the SDK?
https://github.com/amzn/selling-partner-api-docs/blob/main/guides/en-US/developer-guide/SellingPartnerApiDeveloperGuide.md#step-1-request-a-login-with-amazon-access-token

**Comment by bennywoop** [comment, marked as answer]
I found it here:
https://github.com/amzn/selling-partner-api-docs/blob/main/guides/en-US/developer-guide/SellingPartnerApiDeveloperGuide.md#Self-authorization

**Comment by abuzuhri** [comment]
Ok , you welcome any time
